client.go

```diff
@@ -524,25 +525,35 @@ func (ih *internalHandlers) saveApplicationID(_ Session, evt *Ready) {
 //
 //////////////////////////////////////////////////////
 
-func (c *Client) EditInteractionResponse(ctx context.Context, interaction *InteractionCreate, message *UpdateMessage) error {
-	postBody, contentType, err := message.prepare()
-	if err != nil {
-		return err
-	}
+func (c *Client) EditInteractionResponse(ctx context.Context, interaction Interactable, message *UpdateMessage) error {
+	switch interaction.(type) {
+	case *InteractionCreate[MessageComponentInteractionData]:
```

Comment:
Looking at it, I'd say this is a nice use of generics, and I'm up for it. I assume we have to do something similar for the response